common/src/main/java/io/netty/util/AsciiString.java

```diff
@@ -1007,6 +1007,32 @@ public AsciiString toUpperCase() {
 
     /**
      * Copies this string removing white space characters from the beginning and end of the string.
+     * @param c The {@link CharSequence} to trim.
+     * @return a new string with characters {@code <= \\u0020} removed from the beginning and the end.
+     */
+    public static CharSequence trim(CharSequence c) {
+        if (c.getClass() == AsciiString.class) {
```

Comment:
checkNotNull(c)?

Reply:
@windie I think its fine as it will throw directly anyway.